### PR TITLE
HMAN-16 

### PR DESCRIPTION
**Pull request by padmapriyanalam**
*Development in progress*

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/HMAN-16

### Change description ###

HMC Inbound Adapter Endpoint - hearing response message

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by danlysiak — COMMENTED** [review]
A few initial comments

File: src/main/java/uk/gov/hmcts/reform/hmc/client/model/hmi/CftHearingServiceRsp.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CftHearingServiceRsp {
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/client/model/hmi/CftHearingServiceRsp.java:11]:
```suggestion
public class CftHearingServiceResponse {
```

File: src/main/resources/application-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+management:
```
Comment (danlysiak) [inline src/main/resources/application-dev.yaml:1]: We should be able to delete this file now, as we're not using ActiveMQ anywhere (for which this was the original purpose in other repos)

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -16,7 +16,16 @@ spring:
...
+    name: HMC HMI Inbound Adapter
+  jackson:
+    deserialization:
+      fail-on-unknown-properties: true
+    serialization:
+      WRITE_DATES_AS_TIMESTAMPS: false
+
+hmc:
+  cft-hearing-service:
+    host: http://localhost:4459
```
Comment (danlysiak) [inline src/main/resources/application.yaml:28]: Can we define an env var for this too please?
```suggestion
    host: ${CFT_HEARING_SERVICE_URL:http://localhost:4459}
```

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -42,3 +51,7 @@ spring:
 azure:
   application-insights:
     instrumentation-key: ${hmc.AppInsightsInstrumentationKey:00000000-0000-0000-0000-000000000000}
+jms:
+  servicebus:
+    connection-string: ${SB_CONN_STRING:Endpoint=sb://hmc-servicebus-demo.servicebus.windows.net/;SharedAccessKeyName=SendSharedAccessKey;SharedAccessKey=8EY5+1PU55K/yuWM/KEYhdVK549JBqVLXCeIcvHQFm8=;EntityPath=hmc-from-hmi-demo}
```
Comment (danlysiak) [inline src/main/resources/application.yaml:56]: Can we remove this default with real values please, should only be setting this locally as it's sensitive

**Review by danlysiak — COMMENTED** [review]
A few minor comments. We also need to set it up for preview, I think I said I'd help with pointing in that direction 👍

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HmcInboundController.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    public HmcInboundController(HearingManagementService hearingManagementService) {
+        this.hearingManagementService = hearingManagementService;
+    }
+
+    @PutMapping(path = "/listings/{id}")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Response is valid"),
+        @ApiResponse(code = 202, message = "Response is valid"),
+        @ApiResponse(code = 400, message = "Invalid case Id"),
+        @ApiResponse(code = 404, message = "Case Id could not be found"),
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/controllers/HmcInboundController.java]: changed to caseListingID

File: charts/hmc-hmi-inbound-adapter/values.yaml [hunk 6]
```diff
@@ -9,4 +9,7 @@ java:
       secrets:
         - name: AppInsightsInstrumentationKey
           alias: azure.application-insights.instrumentation-key
+        - name: ConnectionString
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply charts/hmc-hmi-inbound-adapter/values.yaml]: changed

File: src/main/java/uk/gov/hmcts/reform/hmc/service/hmi/CftHearingService.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.hmc.service.hmi;
+
+import uk.gov.hmcts.reform.hmc.client.model.hmi.CftHearingServiceRsp;
+
+public interface CftHearingService {
+
+    CftHearingServiceRsp isValidCaseId(String caseId);
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/hmc/service/hmi/CftHearingService.java]: changed to boolean

File: build.gradle [hunk 8]
```diff
@@ -272,12 +274,23 @@ dependencies {
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa'
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-cache'
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-validation'
-
+  implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-openfeign', version: '3.0.2'
+  implementation group: 'io.github.openfeign', name: 'feign-okhttp', version: '11.2'
```
Comment (danlysiak) [inline build.gradle:278]:
```suggestion
```

File: build.gradle [hunk 9]
```diff
@@ -272,12 +274,23 @@ dependencies {
...
   implementation group: 'io.springfox', name: 'springfox-swagger2', version: versions.springfoxSwagger
 
   implementation group: 'uk.gov.hmcts.reform', name: 'logging', version: versions.reformLogging
   implementation group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: versions.reformLogging
 
+  implementation group: 'com.github.tomakehurst', name: 'wiremock', version: '2.27.2'
+  implementation group: 'javax.inject', name: 'javax.inject', version: '1'
+  implementation group: 'com.azure', name: 'azure-messaging-servicebus', version: '7.0.2'
+  implementation group: 'com.azure', name: 'azure-core', version: '1.13.0'
+  implementation group: 'io.hawt', name: 'hawtio-springboot', version: '2.12.0'
```
Comment (danlysiak) [inline build.gradle:288]:
```suggestion
```

File: build.gradle [hunk 10]
```diff
@@ -272,12 +274,23 @@ dependencies {
...
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-validation'
-
+  implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-openfeign', version: '3.0.2'
+  implementation group: 'io.github.openfeign', name: 'feign-okhttp', version: '11.2'
   implementation group: 'io.springfox', name: 'springfox-swagger2', version: versions.springfoxSwagger
 
   implementation group: 'uk.gov.hmcts.reform', name: 'logging', version: versions.reformLogging
   implementation group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: versions.reformLogging
 
+  implementation group: 'com.github.tomakehurst', name: 'wiremock', version: '2.27.2'
```
Comment (danlysiak) [inline build.gradle:284]:
```suggestion
```

File: build.gradle [hunk 11]
```diff
@@ -272,12 +274,23 @@ dependencies {
...
+  implementation group: 'javax.inject', name: 'javax.inject', version: '1'
```
Comment (danlysiak) [inline build.gradle:285]:
```suggestion
```

File: build.gradle [hunk 12]
```diff
@@ -287,6 +300,11 @@ dependencies {
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
 
+  testImplementation group: 'org.mockito', name: 'mockito-inline', version: '3.11.1'
+  testImplementation group: 'org.powermock', name: 'powermock-api-mockito2', version: '2.0.9'
+  testImplementation group: 'org.powermock', name: 'powermock-module-junit4', version: '2.0.9'
+
+  testImplementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-contract-stub-runner', version: '3.0.2'
```
Comment (danlysiak) [inline build.gradle:307]:
```suggestion
```
Not sure if using any of these?

File: build.gradle [hunk 13]
```diff
@@ -297,6 +315,10 @@ dependencies {
 
 mainClassName = 'uk.gov.hmcts.reform.hmc.Application'
 
+run {
+  args = ["--spring.profiles.active=dev"]
+}
+
```
Comment (danlysiak) [inline build.gradle:321]:
```suggestion
```

File: build.gradle (hunk 8 above)
Comment (padmapriyanalam) [reply build.gradle]: removed io.github.openfeign

File: build.gradle (hunk 9 above)
Comment (padmapriyanalam) [reply build.gradle]: removed io.hawt, javax.inject

File: build.gradle (hunk 11 above)
Comment (padmapriyanalam) [reply build.gradle]: removed javax.inject

File: build.gradle (hunk 12 above)
Comment (padmapriyanalam) [reply build.gradle]: removed not used ones

File: build.gradle (hunk 13 above)
Comment (padmapriyanalam) [reply build.gradle]: removed

File: charts/hmc-hmi-inbound-adapter/values.yaml [hunk 14]
```diff
@@ -9,4 +9,8 @@ java:
       secrets:
         - name: AppInsightsInstrumentationKey
           alias: azure.application-insights.instrumentation-key
+        - name: HMCServiceBusConnectionString
+          alias: spring.jms.servicebus.connection-string
```
Comment (danlysiak) [inline charts/hmc-hmi-inbound-adapter/values.yaml:13]:
```suggestion
          alias: jms.servicebus.connection-string
```

**Review by danlysiak — APPROVED** [review]

**Review by danlysiak — APPROVED** [review]

**Review by radhikahollakv — APPROVED** [review]
LGTM

**Pull request merged**